### PR TITLE
Defibrillator get a new mode that can instantly heal a marine to 195 damage. Cost more energy

### PR DESCRIPTION
**Pull request by BraveMole**
## About The Pull Request

Defib gets a new mode, super charge. If you right click instead of left click, you can instantly defib the target.

You need to have medical skill in order to do that, and this will use up 5000 energy.
A normal defib will use 500, and you have a battery of 10k.

You can use your backpack to recharge your defib, but only once since it has a reserve of 10k as well.

## Why It's Good For The Game

Getting more than 200 damage comes down mostly to luck. Corp have no better way of dealing with that than spamming the defib, which can perma you if you are unlucky.

I think being revived or not should only depend on how quickly marines get you, and not if you were unlucky enough to die on a flamed tile or if you took a nade while dead.

If marines managed to get back your corpse, you should be saveable imo.

With this new mode, we keep the normal complexity of reviving, but if someone is too roasted to be resed, the corp can use that.

## Changelog
:cl:
balance: Defibrillator will instantly heal you to 195 damage if you use super charge mode (right click)
balance: Super charge mode consummes 5k energy (out of 10k battery), normal defib uses 500
balance: You can only recharge your defib to full once with your corp backpack (it was 10 times before)
delete: Defib can always be used, no need to open it 
/:cl:

### CODE REVIEW

**Comment by Lumipharon** [comment]
Corps can inaprov and kit a corpse. You can't perma a corpse via defib, but if the heart gets damaged enough they will actually need surgery.

The only way to get someone completely unrevivable is to max out their tox or clone damage.

**Comment by BraveMole** [comment]
> Corps can inaprov and kit a corpse. You can't perma a corpse via defib, but if the heart gets damaged enough they will actually need surgery.
> 
> The only way to get someone completely unrevivable is to max out their tox or clone damage.

they cannot inaprov a corpse, and they can kit only once per limb. So if you have 400, you need to spam defib, which can leads to perma death (unless surgery)

**Comment by goose4429** [comment]
> they cannot inaprov a corpse

They are indeed able to ina the corpse. And RR it, and neura it as well.
Imagine giving work to MO, and not solve every problem via cryo or defib spam.

**Comment by SzkieletorNutric** [comment]
> they cannot inaprov a corpse
Can confirm, Inaprovaline instantly healing a portion of damage somehow works even when the body is dead

**Comment by BraveMole** [comment]
> > they cannot inaprov a corpse
> 
> They are indeed able to ina the corpse. And RR it, and neura it as well. Imagine giving work to MO, and not solve every problem via cryo or defib spam.

then that's a pretty big exploit

**Comment by goose4429** [comment]
I think that if you do both defib and Ina fix, would be cool

**Comment by BraveMole** [comment]
> I think that if you do both defib and Ina fix, would be cool

I mean sure, but ina wouldn't matter with this new defib

**Comment by Losenis** [comment]
I feel that medical has enough mechanical problems to make this understandable. There's some very weird stuff going on that make marines get more damage after death even if it's clearly not intentional, like the time acid sprays would damage corpses or how you can actually damage corpses by punching them normally getting them way past 600 brute. Sometimes corpses on fire would take damage and other times I'd see corpses that were on fire for centuries just have like 20 dmg or so, and random spacetiles would skyrocket the corpse damage to its literal limit. I'd consider this a failsafe to the inevitable bugs that may appear than an actual buff or nerf, since a marine getting 600 dmg after dying at 200 is clearly not intentional and even shipside will struggle to even do anything that isn't deepfrying its heart with defibspam of multiple people.

If marines get over 200 dmg to the point that some call out this affects balance, then something is going extremely wrong. There's already enough things to send a marine shipside that isn't just normal damage numbers.

**Comment by BraveMole** [comment]
> Sometimes corpses on fire would take damage and other times I'd see corpses that were on fire for centuries just have like 20 dmg or so, 
corpse on fire don't take damage, they take damage when on tiles with fire.

**Comment by goose4429** [comment]
> I feel that medical has enough mechanical problems to make this understandable. There's some very weird stuff going on that make marines get more damage after death even if it's clearly not intentional, like the time acid sprays would damage corpses or how you can actually damage corpses by punching them normally getting them way past 600 brute. Sometimes corpses on fire would take damage and other times I'd see corpses that were on fire for centuries just have like 20 dmg or so, and random spacetiles would skyrocket the corpse damage to its literal limit. I'd consider this a failsafe to the inevitable bugs that may appear than an actual buff or nerf, since a marine getting 600 dmg after dying at 200 is clearly not intentional and even shipside will struggle to even do anything that isn't deepfrying its heart with defibspam of multiple people.
> 
> If marines get over 200 dmg to the point that some call out this affects balance, then something is going extremely wrong. There's already enough things to send a marine shipside that isn't just normal damage numbers.

True words, I agree that failsafe value is higher than some possible balance issues in this case

**Comment by DeltaFire15** [comment]
Massive corpsmen buff considering this takes away any pre-defibbing considerations even if command decided to OB a corpse pile.
Most situations with really high damage can be resolved via just kitting + defibbing a few times, whereas if there's massive damage due to some reason (like aforementioned OBs) it can still be fixed by surgery. Having dead bodies in the proximity of combat should influence if you have to be more or less careful around them, besides the fact that otherwise there'd also be zero tradeoff for 'grenade fishing' dead bodies with something like a scoped GL or w/e.
If you are really that much on the fence about people having enough damage to give shipside med a job to do, cap brute / burn damage at something like 400 each instead of just sidestepping revival neccessities.

**Comment by Losenis** [comment]
There's already little to no tradeoff for grenade fishing because the grenade damage is all spread out evenly throughout the body with little threat of fracs over concentrated damage, even if the guy receives 100 brute damage off grenades.

And for every grenade fishing attempt there's another that punched the body further into the maze. This isn't a damage issue, this is a grenade issue.

**Comment by Losenis** [comment]
Not to mention, this wouldn't change much about grenade fishing since due to the aforementioned damage spread that is also both burn and brute, you'd be looking at trauma and health kits reducing the damage of every bodypart twice. Defib damage wouldn't come into play unless something else gave the guy absurd amounts of damage, at which point it also stops being an issue about grenades and instead something else that is most likely not supposed to be doing that to an already dead marine.

**Comment by Ragecrabbo** [comment]
> Getting more than 200 damage comes down mostly to luck. Corp have no better way of dealing with that than spamming the defib, which can perma you if you are unlucky.

Dont sutures exist to deal with this problem? Why not just buff them instead?

**Comment by goose4429** [comment]
New mode is at cool at least

**Comment by ilikesaying** [comment]
> There's already little to no tradeoff for grenade fishing because the grenade damage is all spread out evenly throughout the body with little threat of fracs over concentrated damage, even if the guy receives 100 brute damage off grenades.

pretty sure grenade explosions deal damage to one limb

**Comment by Lumipharon** [comment]
I'm not opposed to this, but this makes it harder to attrition marines due to them being faster to revive, and making it pointless to medivac unless they're missing limbs.

File: code/game/objects/items/defibrillator.dm [hunk 1]
```diff
@@ -220,10 +202,6 @@
 		user.visible_message(span_warning("[icon2html(src, viewers(user))] \The [src] buzzes: Defibrillation failed: Paddles registering >100,000 ohms, Possible cause: Suit or Armor interferring."))
 		return
 
-	var/datum/internal_organ/heart/heart = H.internal_organs_by_name["heart"]
-	if(!issynth(H) && !isrobot(H) && heart && prob(25))
-		heart.take_damage(5) //Allow the defibrilator to possibly worsen heart damage. Still rare enough to just be the "clone damage" of the defib
-
```
Comment (QualityVan) [inline code/game/objects/items/defibrillator.dm:226]: Not in the PR desc or changelog at all. Also don't think it's a needed change given what the pr is adding.

Comment (BraveMole) [reply code/game/objects/items/defibrillator.dm:226]: ah, forgot about this one.

Yeah, I shoudl probably remove

Comment (goose4429) [reply code/game/objects/items/defibrillator.dm:226]: BM subtly buffing marines

**Comment by bowlofcereal** [comment]
MO and CMO gonna be alt tabbing out to watch youtube videos as their main gameplay here soon

**Comment by Autisem** [comment]
Ye this does make the already pretty boring loop of playing MO/CMO that much worse. kinda silly that you can just heal them like that. rechargers exist and I believe the corpsmanbackpack includes are powercell that automaticly recharges defibs so you can staple that onto it aswel(this powercell also is swappable for extra comedic effect)

**Pull request closed without merging**